src/.vitepress/config.ts

```diff
@@ -1,7 +1,28 @@
-import { defineConfig } from "vitepress";
+import { DefaultTheme, UserConfig } from "vitepress";
+import audioPlugin from "./audioPlugin.ts";
+
+/** configをマージする関数。
+ * (config1, config2, config3)という引数で呼び出すと、
+ * `{ extends: { extends: config3, ...config2 }, ...config1 }`のようにマージされる。
+ */
+function mergeConfigs(
+  ...configs: Omit<UserConfig<DefaultTheme.Config>, "extends">[]
+): UserConfig<DefaultTheme.Config> {
+  const base = {} as UserConfig<DefaultTheme.Config>;
+  return configs.reduce((merged, config) => {
+    if (merged === base) {
+      return config;
+    } else {
+      return {
+        ...config,
+        extends: merged,
+      };
+    }
+  }, base);
+}
```

Comment:
よくわかってないけど、最初例外設けるくらいならconfigs[0]をbaseにしたほうが綺麗かもしれない
そうでもないかもしれない

```suggestion
function mergeConfigs(
  ...configs: Omit<UserConfig<DefaultTheme.Config>, "extends">[]
): UserConfig<DefaultTheme.Config> {
  const base = { ...configs[0] } as UserConfig<DefaultTheme.Config>;
  return configs.reduce((merged, config[1:]) => ({
        ...config,
        extends: merged,
  }), base);
}
```

書式がわからないけど。。

まあでも変更面倒だしそのままでも全然良さそう！

Reply:
いい感じにしました。